text/0000-uisize.md

```diff
@@ -0,0 +1,128 @@
+- Feature Name: `uptr_iptr`
+- Start Date: 2016-06-02
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+`isize` and `usize` either must be a `size_t` like type, or `intptr_t` like
+type. Most uses of `usize` and `isize` are as `size_t` or `ptrdiff_t`
+like-types (`Vec<T>`, `[T]`, `.offset`, etc.). Therefore, remove the guarantee
+that `usize` and `isize` are pointer-size, and make them exclusively
+object-sized.
+
+# Motivation
+[motivation]: #motivation
+
+We want to support the embedded space, and new CPUs like the CHERI. These CPUs
+do not support `usize` == `uptr`, and, in the case of the CHERI, don't support
+`uptr` at all. Most CPUs don't actually support the idea that `uptr == usize`:
+just the currently popular ones.
+
+We really want to support usecases where `uptr != usize`. The embedded space,
+where segmented architectures are still very popular, is a very good case for
```

Comment:
Things are moving that way, but a lot of people still use 8-bit architectures; for example, in something like a thermostat, one would use an 8-bit microcontroller.